### PR TITLE
[Documentation] Creating your own Sylius extension

### PR DESCRIPTION
**Pull request by pamil**
| Q               | A
| --------------- | ---
| Bug fix?        | no |
| New feature?    | yes |
| BC breaks?      | no |
| Related tickets | - |
| License         | MIT |

### CODE REVIEW

File: docs/cookbook/creating-extension.rst [hunk 1]
```diff
@@ -0,0 +1,41 @@
+How to create an extension for Sylius?
+======================================
+
+Sylius extension is nothing else but a regular Symfony bundle adding custom behaviour to the default Sylius application.
```
Comment (TheMadeleine) [inline docs/cookbook/creating-extension.rst]: `nothing more`?

File: docs/cookbook/creating-extension.rst [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+
+The best way to create your own extension is to use `Sylius bundle skeleton <https://github.com/Sylius/BundleSkeleton>`_,
+which has built-in infrastructure for designing and testing using Behat.
```
Comment (TheMadeleine) [inline docs/cookbook/creating-extension.rst]: a link to Behat.

`` `Behat <http://behat.org/en/latest/>`_ ``

File: docs/cookbook/creating-extension.rst [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+
+.. note::
+
+    This cookbook assumes that you are worked with Sylius and Symfony before.
```
Comment (TheMadeleine) [inline docs/cookbook/creating-extension.rst]: Oh well. :D I suppose it't too obvious.

File: docs/cookbook/creating-extension.rst [hunk 4]
```diff
@@ -0,0 +1,41 @@
...
+
+**1. Create project using Composer.**
+
+.. code-block:: bash
+
+    $ composer create-project sylius/bundle-skeleton SyliusExtensionPath
+
+**2. Get familiar with basic extension design.**
+
+The skeleton comes with simple application that greets a customer. There are features' scenarios in ``features`` directory;
```
Comment (TheMadeleine) [inline docs/cookbook/creating-extension.rst]: `feature scenarios in the features directory`

File: docs/cookbook/creating-extension.rst [hunk 5]
```diff
@@ -0,0 +1,41 @@
...
+simple example bundle with a controller, a template and a routing configuration in ``src``;
```
Comment (TheMadeleine) [inline docs/cookbook/creating-extension.rst]: `exemplary bundle with a controller, a template and routing configuration`

File: docs/cookbook/creating-extension.rst [hunk 6]
```diff
@@ -0,0 +1,41 @@
...
+and the testing infrastructure in ``tests``.
+
+.. note::
+
+    The ``tests/Application`` directory contains a sample Symfony application built to test your extension against it.
```
Comment (TheMadeleine) [inline docs/cookbook/creating-extension.rst]: against what?

File: docs/cookbook/creating-extension.rst [hunk 7]
```diff
@@ -0,0 +1,41 @@
...
+
+**3. Remove boilerplate files and rename your bundle.**
+
+In most cases you don't want your Sylius extension to greet customer like it is now, so feel free to remove unnecessary
+controllers, assets and feature. You will also want to change the extension namespace from ``Acme\ExampleBundle`` to a
```
Comment (TheMadeleine) [inline docs/cookbook/creating-extension.rst]: feature*s*.

`You may also want to change the extension's namespace`

File: docs/cookbook/creating-extension.rst [hunk 8]
```diff
@@ -0,0 +1,41 @@
...
+more meaningful one. Keep in mind that these changes also need to be done in ``tests/Application`` and ``composer.json``.
+
+**4. Implement your awesome features.**
+
+Looking at existing Sylius extensions like `Lakion\SyliusCmsBundle`_ or `Lakion\SyliusElasticSearchBundle`_ are great
```
Comment (TheMadeleine) [inline docs/cookbook/creating-extension.rst]: `` Looking at existing Sylius extensions like `Lakion\SyliusCmsBundle`_ or `Lakion\SyliusElasticSearchBundle`_ is a great way to start developing your own extensions.``

File: docs/cookbook/creating-extension.rst (hunk 3 above)
Comment (pamil) [inline docs/cookbook/creating-extension.rst]: 🌮

File: docs/cookbook/creating-extension.rst [hunk 9]
```diff
@@ -0,0 +1,37 @@
+How to create an extension for Sylius?
+======================================
+
+Sylius extension is nothing more but a regular Symfony bundle adding custom behaviour to the default Sylius application.
+
+The best way to create your own extension is to use `Sylius bundle skeleton <https://github.com/Sylius/BundleSkeleton>`_,
+which has built-in infrastructure for designing and testing using `Behat <http://behat.org/en/latest/>`_.
+
+**1. Create project using Composer.**
```
Comment (TheMadeleine) [inline docs/cookbook/creating-extension.rst]: I was using another "style" in the cookbooks already:
```
1. Create project using Composer.
----------------------------------
```

**Comment by pjedrzejewski** [comment]
I am missing the following things:

* Links/references to BDD with Behat and phpspec + PHPUnit (that you can use them all)
* Clear explanation that the extension can be created anywhere, not inside the Sylius app, because it contains the test application inside already. I know it is obvious to us, but not for the newcomers.

**Comment by pjedrzejewski** [comment]
Thanks Kamil! :)

**Pull request merged**